server/devpi_server/importexport.py

```diff
@@ -35,8 +53,8 @@ def do_import(path, xom):
     if not path.check():
         fatal("path for importing not found: %s" %(path))
 
-    with xom.keyfs.transaction():
-        if not xom.model.is_empty():
+    with xom.keyfs.transaction(write=False):
+        if not no_existing_data(xom):
```

Comment:
Nice move to use a function.
I would name it `has_users_or_stages` (referring to the error message in the exception) and thus avoid the double negation.